kratos/processes/compute_nodal_gradient_process.cpp

```diff
@@ -337,13 +385,10 @@ array_1d<double, 3>& ComputeNodalGradientProcess<ComputeNodalGradientProcessSett
 template <>
 void ComputeNodalGradientProcess<ComputeNodalGradientProcessSettings::SaveAsHistoricalVariable>::PonderateGradient()
 {
-    const auto it_node_begin = mrModelPart.NodesBegin();
-
-    #pragma omp parallel for
-    for(int i = 0; i < static_cast<int>(mrModelPart.Nodes().size()); ++i) {
-        auto it_node = it_node_begin + i;
-        it_node->FastGetSolutionStepValue(*mpGradientVariable) /= it_node->GetValue(*mpAreaVariable);
-    }
+    block_for_each(mrModelPart.Nodes(), [&](Node<3>& rNode){
```

Comment:
Make a PR only for the parallel utilities and I will approve easily. Later we can discuss the other changes